### PR TITLE
Maintain Chapter 5

### LINKED ISSUE #561: Maintain Chapter 5 Using structs to structure related data
- [ ] In 5-2, wrong output of the program, does not print what it should print with println!
- [ ] In 5-2, print() still used in the last example, should we remove it ? but we need to implement PrintTrait manually so we should probably leave it
- [ ] In 5-3, dead link  [Chapter8][Modules]

### PR DESCRIPTION
**Pull request by TAdev0**
Resolves #561 

@enitrat  just so you know, listings still have issues.
There were many inconsistencies in this chapter. I manually modified so that the order of ''quoted'' listings is consistent when reading the book.
But many listings of chapter 5 started with 04_... , and the are missing numbers with gaps, and it seems all this was not detected by the script.
i think also that quoting as different listings the same listing file (but including only some part of the file each time) might cause issues with the script, idk

- - -
This change is [Reviewable](https://reviewable.io/reviews/cairo-book/cairo-book/574)

### CODE REVIEW

**Comment by enitrat** [comment]
> But many listings of chapter 5 started with 04_... , and the are missing numbers with gaps, and it seems all this was not detected by the script.

Yeah probably because they were in the correct order, the script doesn't check if the Listing number matches the file number as you could have different listings referring to the same file

File: src/ch05-02-an-example-program-using-structs.md [hunk 1]
```diff
@@ -63,39 +62,49 @@ Here we’ve defined a struct and named it `Rectangle`. Inside the curly bracket
 
 ## Adding Useful Functionality with Trait
 
-It’d be useful to be able to print an instance of `Rectangle` while we’re debugging our program and see the values for all its fields. Listing 5-9 tries using the `print` as we have used in previous chapters. This won’t work.
+It’d be useful to be able to print an instance of `Rectangle` while we’re debugging our program and see the values for all its fields. Listing 5-9 tries to print `rectangle` using the `print` function of the `PrintTrait`. This won’t work.
```
Comment (enitrat) [inline src/ch05-02-an-example-program-using-structs.md:65]: Can you create an issue to refactor this section by completely removing the print trait? Now that we have debug and `println` macro it doesn't make sense.

**Review by enitrat — DISMISSED** [review]

File: src/ch05-02-an-example-program-using-structs.md (hunk 1 above)
Comment (TAdev0) [reply src/ch05-02-an-example-program-using-structs.md]: @enitrat  i just modified this whole section, tell me if this is ok, or if you prefer that i create an issue

Comment (enitrat) [reply src/ch05-02-an-example-program-using-structs.md]: in a new issue / PR please

Comment (TAdev0) [reply src/ch05-02-an-example-program-using-structs.md]: ok!

File: src/ch05-02-an-example-program-using-structs.md [hunk 2]
```diff
@@ -60,9 +60,9 @@ We use structs to add meaning by labeling the data. We can transform the tuple w
 
 Here we’ve defined a struct and named it `Rectangle`. Inside the curly brackets, we defined the fields as `width` and `height`, both of which have type `u64`. Then, in `main`, we created a particular instance of `Rectangle` that has a width of `30` and a height of `10`. Our `area` function is now defined with one parameter, which we’ve named `rectangle` which is of type `Rectangle` struct. We can then access the fields of the instance with dot notation, and it gives descriptive names to the values rather than using the tuple index values of `0` and `1`.
 
-## Adding Useful Functionality with Trait
+## Adding Uselful Functionality with Trait
```
Comment (enitrat) [inline src/ch05-02-an-example-program-using-structs.md:63]:
```suggestion
## Adding Useful Functionality with Trait
```

File: src/ch05-01-defining-and-instantiating-structs.md [hunk 3]
```diff
@@ -7,30 +7,30 @@ To define a struct, we enter the keyword `struct` and name the entire struct. A
...
+<span class="caption">Listing 5-2: Creating two instances of the `User` struct</span>
 
-To get a specific value from a struct, we use dot notation. For example, to access this user’s email address, we use `user1.email`. If the instance is mutable, we can change a value by using the dot notation and assigning into a particular field. Listing 5-3 shows how to change the value in the `email` field of a mutable `User` instance.
+To get a specific value from a struct, we use dot notation. For example, to access `user1`'s email address, we use `user1.email`. If the instance is mutable, we can change a value by using the dot notation and assigning into a particular field. Listing 5-3 shows how to change the value in the `email` field of a mutable `User` instance.
 
 <span class="filename">Filename: src/lib.cairo</span>
 
 ```rust
-{{#rustdoc_include ../listings/ch05-using-structs-to-structure-related-data/listing_04_03_mut_struct/src/lib.cairo:main}}
+{{#rustdoc_include ../listings/ch05-using-structs-to-structure-related-data/listing_05_02_mut_struct/src/lib.cairo:main}}
```
Comment (enitrat) [inline src/ch05-01-defining-and-instantiating-structs.md:33]:
```suggestion
{{#rustdoc_include ../listings/ch05-using-structs-to-structure-related-data/listing_05_03_mut_struct/src/lib.cairo:main}}
```

should be, no?

File: src/ch05-02-an-example-program-using-structs.md [hunk 4]
```diff
@@ -63,39 +62,50 @@ Here we’ve defined a struct and named it `Rectangle`. Inside the curly bracket
...
-It’d be useful to be able to print an instance of `Rectangle` while we’re debugging our program and see the values for all its fields. Listing 5-9 tries using the `print` as we have used in previous chapters. This won’t work.
+It’d be useful to be able to print an instance of `Rectangle` while we’re debugging our program and see the values for all its fields. Listing 5-9 tries to print `rectangle` using the `print` function. This won’t work.
 
 <span class="filename">Filename: src/lib.cairo</span>
 
 <!-- TODO implement debug instead -->
 
 ```rust
-{{#include ../listings/ch05-using-structs-to-structure-related-data/listing_04_10_print_rectangle/src/lib.cairo:here}}
+{{#include ../listings/ch05-using-structs-to-structure-related-data/listing_05_10_print_rectangle/src/lib.cairo:here}}
```
Comment (enitrat) [inline src/ch05-02-an-example-program-using-structs.md:72]:
```suggestion
{{#include ../listings/ch05-using-structs-to-structure-related-data/listing_05_09_print_rectangle/src/lib.cairo:here}}
```

File: src/ch05-02-an-example-program-using-structs.md [hunk 5]
```diff
@@ -63,39 +62,50 @@ Here we’ve defined a struct and named it `Rectangle`. Inside the curly bracket
...
 
-The `print` trait is implemented for many data types, but not for the `Rectangle` struct. We can fix this by implementing the `PrintTrait` trait on `Rectangle` as shown in Listing 5-10.
-To learn more about traits, see [Traits in Cairo](ch08-02-traits-in-cairo.md).
+The `PrintTrait` trait is implemented for many data types, but not for the `Rectangle` struct. We can fix this by implementing the `PrintTrait` trait on `Rectangle` as shown in Listing 5-10. To learn more about traits, see [Traits in Cairo](ch08-02-traits-in-cairo.md).
 
 <span class="filename">Filename: src/lib.cairo</span>
 
 ```rust
-{{#rustdoc_include ../listings/ch05-using-structs-to-structure-related-data/listing_04_10_print_rectangle/src/lib.cairo:all}}
+{{#rustdoc_include ../listings/ch05-using-structs-to-structure-related-data/listing_05_10_print_rectangle/src/lib.cairo:all}}
```
Comment (enitrat) [inline src/ch05-02-an-example-program-using-structs.md:95]:
```suggestion
{{#rustdoc_include ../listings/ch05-using-structs-to-structure-related-data/listing_05_09_print_rectangle/src/lib.cairo:all}}
```

unless it's actually the one above that's wrong?

Comment (TAdev0) [reply src/ch05-02-an-example-program-using-structs.md:95]: yes, i modified at, at there was a listing 01 and a listing 3 without 2, i changed it to 2 manually. 
I checked with the server, everything is ok

Comment (TAdev0) [reply src/ch05-02-an-example-program-using-structs.md:95]: well i can change for listing 9, i created the one i deleted for the struct print and named it listing 09, will change that for consistency

File: src/ch05-01-defining-and-instantiating-structs.md (hunk 3 above)
Comment (TAdev0) [reply src/ch05-01-defining-and-instantiating-structs.md:33]: yes, i modified that, as there was a listing 01 and a listing 3 without 2, i changed it to 2 manually.
I checked with the server, everything is ok

**Review by enitrat — APPROVED** [review]

**Pull request merged**